grace/utils/augment_graph.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Dict, Union, Tuple
+
+import torch
+
+import numpy as np
+
+from grace.base import GraphAttrs, Annotation
+
+
+class RandomEdgeAdditionAndRemoval:
+    """Rotate the image and graph in tandem; i.e., the graph x-y coordinates
+    will be transformed to reflect the image rotation.
+
+    Accepts an image stack of size (C,W,H) or (B,C,W,H) and a dictionary
+    which includes the graph object.
+
+    Parameters
+    ----------
+    n_add : int
+        Number of new edges to add
+    n_remove : Tuple[int]
+        Number of existing edges to remove
+    rng : numpy.random.Generator
+        Random number generator
+    """
+
+    def __init__(
+        self,
+        n_add: int = 10,
+        n_remove: int = 10,
+        rng: np.random.Generator = np.random.default_rng(),
+    ):
+        self.rng = rng
+        self.n_add = n_add
+        self.n_remove = n_remove
+
+    def __call__(
+        self, x: torch.Tensor, graph: Dict[Union[str, GraphAttrs], Any]
```

Comment:
I think this type annotation can be simplified, given that you're passing the target dictionary:

```suggestion
        self, x: torch.Tensor, graph: Dict[str, Any]
```